---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# scClustAnnot

<!-- badges: start -->

<!-- badges: end -->

An R package integrate into [Seurat V5](https://github.com/satijalab/seurat), meant to aid clustering and annotation step in scRNA-seq analysis

## Description

`scClustAnnot` is an R package to perform clustering and annotation step in scRNA-seq work flow. Currently, clustering and annotation in scRNA-seq analysis requires many repetitive manual works, including resolution tuning, identifying differentially expressed genes (DEGs), and interpreting DEGs biological functions at given context, which is time consuming and subject to researcher’s bias. There exists some automatic annotation tool, however, mostly done by map to previously annotated data. In that case, the quality of previously annotated data is a great concern. This package aims to automate above manual task to an extent, so that researchers’ bias can be reduced and analysis can be performed more effectively. The `scClustAnnot` package was developed using `R version 4.5.0 (2025-04-11 ucrt)`, platform `Microsoft Windows 11 25H2 (64 bit)`

## Installation

You can install the development version of scClustAnnot like so:

``` r
> install.packages("devtools")
> library("devtools")
> devtools::install_github("DDMMMAA/scClustAnnot", build_vignettes = TRUE)
> library("scClustAnnot")
```

## Overview

...

## Contributions

...

## References

...

## Acknowledgements

...

## Example

This is a basic example which shows you how to solve a common problem:

```{r example}
library(scClustAnnot)
## basic example code
```

What is special about using `README.Rmd` instead of just `README.md`? You can include R chunks like so:

```{r cars}
summary(cars)
```

You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this.

You can also embed plots, for example:

```{r pressure, echo = FALSE}
plot(pressure)
```

In that case, don't forget to commit and push the resulting figure files, so they display on GitHub and CRAN.
